pkg/addonmanager/constants/constants.go

```diff
@@ -33,9 +35,80 @@ const (
+
+	// // HostingClusterManifestLabel is the annotation for indicating the manifest should be deployed on the
+	// // hosting cluster
+	// HostingClusterManifestLabel = "addon.open-cluster-management.io/hosting-cluster-manifest"
+
+	InstallModeBuiltinValueKey = "InstallMode"
+	InstallModeHosted          = "Hosted"
+	InstallModeDefault         = "Default"
+
+	// HostingClusterNameAnnotation is the annotation for indicating the hosting cluster name
+	HostingClusterNameAnnotation = "addon.open-cluster-management.io/hosting-cluster-name"
+
+	// HostedManifestLocationLabel is the label for indicating where the manifest should be deployed in Hosted mode
+	HostedManifestLocationLabel = "addon.open-cluster-management.io/hosted-manifest-location"
+
+	// HostedManifestLocationManaged indicates the manifest will be deployed on the managed cluster in Hosted mode, it
+	// is the default value of a manifest in Hosted mode
+	HostedManifestLocationManaged = "managed"
+	// HostedManifestLocationHosting indicates the manifest will be deployed on the hosting cluster in Hosted mode
+	HostedManifestLocationHosting = "hosting"
+	// HostedManifestLocationNone indicates the manifest will not be deployed in Hosted mode
+	HostedManifestLocationNone = "none"
+
+	// HostingManifestFinalizer is the finalizer for an addon which has deployed manifests on the external
+	// hosting cluster in Hosted mode
+	HostingManifestFinalizer = "cluster.open-cluster-management.io/hosting-manifests-cleanup"
+
+	// AddonHostingManifestApplied is a condition type representing whether the manifest of an addon
+	// is applied on the hosting cluster correctly.
+	AddonHostingManifestApplied = "HostingManifestApplied"
+
+	// HostingClusterValid is a condition type representing whether the hosting cluster is valid in Hosted mode
+	HostingClusterValidity = "HostingClusterValidity"
+
+	// HostingClusterValidityReasonValid is the reason of condition HostingClusterValidity indicating the hosting
+	// cluster is valid
+	HostingClusterValidityReasonValid = "HostingClusterValid"
+
+	// HostingClusterValidityReasonInvalid is the reason of condition HostingClusterValidity indicating the hosting
+	// cluster is invalid
+	HostingClusterValidityReasonInvalid = "HostingClusterInvalid"
 )
 
 // DeployWorkName return the name of work for the addon
 func DeployWorkName(addonName string) string {
 	return fmt.Sprintf("addon-%s-deploy", addonName)
 }
+
+// DeployHostingWorkName return the name of manifest work on hosting cluster for the addon
+func DeployHostingWorkName(addonNamespace, addonName string) string {
+	return fmt.Sprintf("%s-hosting-%s", DeployWorkName(addonName), addonNamespace)
+}
+
+// GetHostedModeInfo returns addon installation mode and hosting cluster name.
+func GetHostedModeInfo(annotations map[string]string) (string, string) {
+	hostingClusterName, ok := annotations[HostingClusterNameAnnotation]
+	if !ok {
+		return InstallModeDefault, ""
+	}
+
+	return InstallModeHosted, hostingClusterName
```

Comment:
what if hostingClusterName is empty?

Reply:
I think then an invalid hosting cluster condition will be added to the managedclusteraddon